### PR TITLE
add secret masker for trimming double qoutes.

### PR DESCRIPTION
**Pull request by TingluoHuang**
https://github.com/github/c2c-actions-runtime/issues/451

### CODE REVIEW

File: src/Sdk/DTLogging/Logging/ValueEncoders.cs [hunk 1]
```diff
@@ -60,6 +60,20 @@ public static String XmlDataEscape(String value)
             return SecurityElement.Escape(value);
         }
 
+        public static String TrimDoubelQuoteDataEscape(String value)
+        {
+            var trimmed = string.Empty;
+            if (!string.IsNullOrEmpty(value) &&
+                value.StartsWith('"') &&
+                value.EndsWith('"') &&
+                value.Length > 2)
```
Comment (TingluoHuang) [inline src/Sdk/DTLogging/Logging/ValueEncoders.cs:69]: do we need min length for the string?

Comment (ericsciple) [reply src/Sdk/DTLogging/Logging/ValueEncoders.cs:69]: I would. If it's less than N then it's not a very good secret... maybe 5 or 6?

Comment (ericsciple) [reply src/Sdk/DTLogging/Logging/ValueEncoders.cs:69]: So 5 or 6, plus 2 more for the surrounding quotes.

File: src/Sdk/DTLogging/Logging/ValueEncoders.cs [hunk 2]
```diff
@@ -60,6 +60,20 @@ public static String XmlDataEscape(String value)
             return SecurityElement.Escape(value);
         }
 
+        public static String TrimDoubelQuoteDataEscape(String value)
```
Comment (ericsciple) [inline src/Sdk/DTLogging/Logging/ValueEncoders.cs:63]: How about `TrimQuotes` or `TrimDoubleQuotes`?

"Data" is terminology specific to URI, to distinguish between a segment within the URI vs the whole thing.

Comment (ericsciple) [reply src/Sdk/DTLogging/Logging/ValueEncoders.cs:63]: And "Escape" terminology doesnt apply here either

File: src/Sdk/DTLogging/Logging/ValueEncoders.cs [hunk 3]
```diff
@@ -60,6 +60,20 @@ public static String XmlDataEscape(String value)
...
+            var trimmed = string.Empty;
+            if (!string.IsNullOrEmpty(value) &&
+                value.StartsWith('"') &&
+                value.EndsWith('"') &&
+                value.Length > 2)
+            {
+                trimmed = value.Substring(1, value.Length - 2);
+            }
+
+            return string.IsNullOrEmpty(trimmed) ? string.Empty : trimmed;
```
Comment (ericsciple) [inline src/Sdk/DTLogging/Logging/ValueEncoders.cs:74]: why not just return `trimmed` :)

File: src/Sdk/DTLogging/Logging/ValueEncoders.cs (hunk 2 above)
Comment (TingluoHuang) [reply src/Sdk/DTLogging/Logging/ValueEncoders.cs:63]: 👍

**Pull request merged**